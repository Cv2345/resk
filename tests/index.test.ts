import {
  extractGists,
  loadFile,
  flatten,
  noneFormatter,
  objectFromEntries,
} from '../src/'
import * as path from 'path'

describe('languages:', () => {
  /* Language gists extracts. */
  test('typescript', async () => {
    const file = path.resolve(__dirname, './__fixtures__/typescript.ts')
    const gists = extractGists(loadFile(file)!)
    expect(gists).toMatchSnapshot()
    expect(gists.length).toBe(3)
  })

  test('javascript', async () => {
    const file = path.resolve(__dirname, './__fixtures__/javascript.js')
    const gists = extractGists(loadFile(file)!)
    expect(gists).toMatchSnapshot()
    expect(gists.length).toBe(3)
  })
})

test('loads the file correctly', async () => {
  const file = path.resolve(__dirname, './__fixtures__/typescript.ts')
  expect(loadFile(file)).toMatchSnapshot()
})

test('flattens', () => {
  expect(
    flatten([
      [1, 2, 3],
      [4, 5, 6],
    ]),
  ).toEqual([1, 2, 3, 4, 5, 6])
})

test('noneFormatter', () => {
  const rand = Math.random().toString()
  expect(noneFormatter(rand)).toBe(rand)
})

test('objectFromEntries', () => {
  expect(
    objectFromEntries([
      ['foo', '1'],
      ['bar', '2'],
    ]),
  ).toEqual({
    foo: '1',
    bar: '2',
  })
})
