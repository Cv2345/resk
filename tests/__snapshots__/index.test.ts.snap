// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads the file correctly 1`] = `
Object {
  "extension": ".ts",
  "lang": Object {
    "end": /\\\\/\\\\\\*\\\\s\\*resk end\\\\s\\*\\\\\\*\\\\//,
    "extensions": Array [
      ".ts",
      ".tsx",
    ],
    "formatter": [Function],
    "gistter": [Function],
    "start": /\\\\/\\\\\\*\\\\s\\*resk start\\\\s\\+\\\\"\\(\\.\\+\\)\\\\"\\\\s\\*\\\\\\*\\\\//,
  },
  "source": "/* resk start \\"schema\\" */
const typeDefs = \`
  type Query {
    frontPage: [Fruit!]!
    fruits: [Fruit!]!
    customers: [Customer!]!
  }
  type Mutation {
    addFruitToBasket: Boolean!
  }
  type Fruit {
    name: String!
    count: Int!
  }
  type Customer {
    id: ID!
    basket: [Fruit!]!
  }
\`
/* resk end */

const resolvers = {
  Query: {
    frontPage: () => [
      { name: 'orange', count: 10 },
      { name: 'apple', count: 1 },
    ],
    fruits: () => {
      /* resk start \\"fruits\\" */
      const fruits = [
        { name: 'orange', count: 10 },
        { name: 'apple', count: 1 },
        { name: 'strawberries', count: 100 },
      ]
      /* resk end */

      return fruits
    },
    customers: () => [
      { id: 1, basket: [{ name: 'orange', count: 1 }] },
      { id: 2, basket: [{ name: 'apple', count: 2 }] },
    ],
  },
  Mutation: {
    addFruitToBasket: () => true,
  },
}

// Auth
/* resk start \\"users\\" */
const users = {
  mathew: {
    id: 1,
    name: 'Mathew',
    role: 'admin',
  },
  george: {
    id: 2,
    name: 'George',
    role: 'editor',
  },
  johnny: {
    id: 3,
    name: 'Johnny',
    role: 'customer',
  },
}
/* resk end */

function getUser(req) {
  const auth = req.get('Authorization')
  if (users[auth]) {
    return users[auth]
  } else {
    return null
  }
}
",
}
`;

exports[`resk: test languages 1`] = `
Object {
  "branch": "master",
  "content": "{\\"schema.ts\\":\\"https://api.github.com/gists/schema.ts\\",\\"fruits.ts\\":\\"https://api.github.com/gists/fruits.ts\\",\\"users.ts\\":\\"https://api.github.com/gists/users.ts\\",\\"schema.js\\":\\"https://api.github.com/gists/schema.js\\",\\"fruits.js\\":\\"https://api.github.com/gists/fruits.js\\",\\"users.js\\":\\"https://api.github.com/gists/users.js\\",\\"action.yml\\":\\"https://api.github.com/gists/action.yml\\",\\"nested.yml\\":\\"https://api.github.com/gists/nested.yml\\"}",
  "message": "Resk action paths update.",
}
`;

exports[`resk: test languages 2`] = `
Object {
  "action.yml": "steps:
  - test: \\"yarn run test\\"
  - publish:
      on: npm
      package: resk
",
  "fruits.js": "const fruits = [
  { name: 'orange', count: 10 },
  { name: 'apple', count: 1 },
  { name: 'strawberries', count: 100 },
]
",
  "fruits.ts": "const fruits = [
  { name: 'orange', count: 10 },
  { name: 'apple', count: 1 },
  { name: 'strawberries', count: 100 },
]
",
  "nested.yml": "- 1
- 2
- 3
",
  "schema.js": "const typeDefs = \`
  type Query {
    frontPage: [Fruit!]!
    fruits: [Fruit!]!
    customers: [Customer!]!
  }
  type Mutation {
    addFruitToBasket: Boolean!
  }
  type Fruit {
    name: String!
    count: Int!
  }
  type Customer {
    id: ID!
    basket: [Fruit!]!
  }
\`
",
  "schema.ts": "const typeDefs = \`
  type Query {
    frontPage: [Fruit!]!
    fruits: [Fruit!]!
    customers: [Customer!]!
  }
  type Mutation {
    addFruitToBasket: Boolean!
  }
  type Fruit {
    name: String!
    count: Int!
  }
  type Customer {
    id: ID!
    basket: [Fruit!]!
  }
\`
",
  "users.js": "const users = {
  mathew: {
    id: 1,
    name: 'Mathew',
    role: 'admin',
  },
  george: {
    id: 2,
    name: 'George',
    role: 'editor',
  },
  johnny: {
    id: 3,
    name: 'Johnny',
    role: 'customer',
  },
}
",
  "users.ts": "const users = {
  mathew: {
    id: 1,
    name: 'Mathew',
    role: 'admin',
  },
  george: {
    id: 2,
    name: 'George',
    role: 'editor',
  },
  johnny: {
    id: 3,
    name: 'Johnny',
    role: 'customer',
  },
}
",
}
`;
